buildenv/jenkins/common/variables-functions.groovy

```diff
@@ -702,19 +702,151 @@ def set_slack_channel() {
-        echo "Artifactory Manual Cleanup: ${env.ARTIFACTORY_MANUAL_CLEANUP}"
+    ARTIFACTORY_CONFIG = [:]
+    echo "Configure Artifactory..."
+
+    if (VARIABLES.artifactory.defaultGeo) {
+        // Allow default geo to be overridden with a param. Used by the Clenaup script to target a specific server.
+        ARTIFACTORY_CONFIG['defaultGeo'] = (params.ARTIFACTORY_GEO) ? params.ARTIFACTORY_GEO : VARIABLES.artifactory.defaultGeo
+        ARTIFACTORY_CONFIG['geos'] = VARIABLES.artifactory.server.keySet()
+        ARTIFACTORY_CONFIG['repo'] = VARIABLES.artifactory.repo
+        ARTIFACTORY_CONFIG['uploadDir'] = "${ARTIFACTORY_CONFIG['repo']}/${JOB_NAME}/${BUILD_ID}/"
+        //println VARIABLES.artifactory
+        //println ARTIFACTORY_CONFIG['defaultServerGeo']
+        //println ARTIFACTORY_CONFIG['geos']
+        //println ARTIFACTORY_CONFIG['repo']
+        //println ARTIFACTORY_CONFIG['manualCleanup']
+
+        for (geo in ARTIFACTORY_CONFIG['geos']) {
+            ARTIFACTORY_CONFIG[geo] = [:]
+            ARTIFACTORY_CONFIG[geo]['server'] = get_value(VARIABLES.artifactory.server, geo)
+            ARTIFACTORY_CONFIG[geo]['numArtifacts'] = get_value(VARIABLES.artifactory.numArtifacts, geo).toInteger()
+            ARTIFACTORY_CONFIG[geo]['daysToKeepArtifacts'] = get_value(VARIABLES.artifactory.daysToKeepArtifacts, geo).toInteger()
+            ARTIFACTORY_CONFIG[geo]['manualCleanup'] = get_value(VARIABLES.artifactory.manualCleanup, geo)
+            ARTIFACTORY_CONFIG[geo]['vpn'] = get_value(VARIABLES.artifactory.vpn, geo)
+            //println geo
+            //println ARTIFACTORY_CONFIG[geo]['server']
+            //println ARTIFACTORY_CONFIG[geo]['numArtifacts']
+            //println ARTIFACTORY_CONFIG[geo]['daysToKeepArtifacts']
+        }
+        ARTIFACTORY_CONFIG[VARIABLES.artifactory.defaultGeo]['uploadBool'] = true
+
+        // Determine if we need to upload more than the default server
+        if (ARTIFACTORY_CONFIG['geos'].size() > 1) {
+            // What platform did we build on
+            compilePlatform = get_node_platform(NODE_LABELS)
+
+            // See if there are servers with colocated nodes of matching platform
+            def testNodes = jenkins.model.Jenkins.instance.getLabel('ci.role.test').getNodes()
+            //println testNodes
+            for (node in testNodes) {
+                def nodeGeo = get_node_geo(node.getLabelString())
+                def nodePlatform = get_node_platform(node.getLabelString())
+                //println nodeGeo
+                //println nodePlatform
+                // Upload if there is a server at geo where there are machines matching our platform.
+                if (nodePlatform == compilePlatform && ARTIFACTORY_CONFIG['geos'].contains(nodeGeo)) {
+                    ARTIFACTORY_CONFIG[nodeGeo]['uploadBool'] = true
+                    //println ARTIFACTORY_CONFIG[nodeGeo]['uploadBool']
```

Comment:
There are multiple nodes per geo. It should set the flag only once (and skip the other nodes with same geo):
```
if (!ARTIFACTORY_CONFIG[nodeGeo].contains('uploadBool') {
    ARTIFACTORY_CONFIG[nodeGeo]['uploadBool'] = true
}
```